src/eth/primitives/execution.rs

```diff
@@ -57,55 +56,70 @@ impl Execution {
         None
     }
 
-    /// Check if the current transaction was completed normally.
+    /// Checks if the current transaction was completed normally.
     pub fn is_success(&self) -> bool {
         matches!(self.result, ExecutionResult::Success { .. })
     }
 
-    pub fn cmp_with_receipt(&self, receipt: &ExternalReceipt) -> anyhow::Result<()> {
-        // Should the gas be the same?
-        // assert_eq!(execution.gas, receipt.gas_used.unwrap_or_default().into());
-
-        let exec_result: U64 = match self.result {
-            ExecutionResult::Success => 1,
-            _ => 0,
-        }
-        .into();
-        let rcpt_status = receipt.status.unwrap_or_default();
-        if exec_result != rcpt_status {
-            return Err(anyhow!("tx result mismatch, expected: {:?} got: {:?}", rcpt_status, exec_result));
+    /// Checks if current execution state matches the information present in the external receipt.
+    pub fn compare_with_receipt(&self, receipt: &ExternalReceipt) -> anyhow::Result<()> {
+        // compare execution status
+        if self.is_success() != receipt.is_success() {
+            return log_and_err!(format!(
+                "transaction status does not match | hash={} execution={:?} receipt={:?}",
+                receipt.hash(),
+                self.result,
+                receipt.status
+            ));
         }
 
+        // compare logs length
         if self.logs.len() != receipt.logs.len() {
-            return Err(anyhow!(
-                "tx logs length mismatch, expected: {:?} got: {:?}",
-                receipt.logs.len(),
-                self.logs.len()
+            return log_and_err!(format!(
+                "logs length does not match | hash={} execution={} receipt={}",
```

Comment:
I think here the correct would be "do not", since the subject is logs which is not 3rd-person singular. Same everywhere else where the subject is plural. https://testbook.com/key-differences/difference-between-do-and-does